google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -155,6 +154,17 @@ public void customize(
     editAppYamlButton.addActionListener(event -> openModuleSettings());
     dockerfileDirectoryPathLink.addHyperlinkListener(event -> openModuleSettings());
 
+    // we need to add a separate state change listener for project selector since hyperlinks are not
+    // caught by standard settings editor watcher.
```

Comment:
what about the other potential solution for this that you mentioned? (the platform interface which will expose this to the editor watcher)

Reply:
also, if we do go with this, can this be promoted to the common `AppEngineDeploymentConfigurationPanel` since this logic is duplicated for flex and standard

Reply:
It's called `UserActivityProviderComponent`, it must be supported, I tried it but it didn't work from the initial tries - they traverse container hierarchy and I think something is missing again. It will take more time than needed to debug this.

Good idea, I can put that checkbox into general panel.

Reply:
sounds good. lets explore that separately then